paddle/fluid/operators/pool_op.cc

```diff
@@ -162,6 +159,24 @@ framework::OpKernelType PoolOp::GetExpectedKernelType(
       layout_, library_);
 }
 
+framework::OpKernelType PoolOp::GetKernelTypeForVar(
+    const std::string& var_name, const Tensor& tensor,
+    const framework::OpKernelType& expected_kernel_type) const {
+#ifdef PADDLE_WITH_MKLDNN
+  if ((expected_kernel_type.data_layout_ == framework::DataLayout::kMKLDNN) &&
+      (tensor.layout() != framework::DataLayout::kMKLDNN)) {
+    auto attrs = Attrs();
```

Comment:
This code is exactly the same as in other NHWC operators. Can't it be moved to OperatorWithKernel class?

Reply:
It will be a bit different in conv & conv_transpose :) I will consider making a common code once I have all 4 ops supporting NHWC (next PR).